plugins/woocommerce-admin/client/marketplace/components/content/content.tsx

```diff
@@ -212,30 +340,46 @@ export default function Content(): JSX.Element {
 		recordLegacyTabView( marketplaceViewProps );
 	}, [ query?.tab, query?.term, query?.section, query?.category ] );
 
+	// Reset current page when tab, term, or category changes
+	useEffect( () => {
+		setCurrentPage( 1 );
+		setFirstNewProductId( 0 );
+	}, [ selectedTab, query?.category, query?.term ] );
+
+	const getProductType = ( tab: string ): ProductType => {
+		switch ( tab ) {
+			case 'themes':
+				return ProductType.theme;
+			case 'business-services':
+				return ProductType.businessService;
+			default:
+				return ProductType.extension;
+		}
+	};
+
+	useEffect( () => {
```

Comment:
This is beautiful.

```suggestion
	// Maintain product focus for accessibility.
	useEffect( () => {
```

Reply:
added in https://github.com/woocommerce/woocommerce/pull/51434/commits/a26a03d7e68fc32525e2e1029e14d5319d58e8b7